tests/dummy/app/components/login-form.js

```diff
@@ -6,15 +6,18 @@ export default Component.extend({
   session: service('session'),
 
   actions: {
-    authenticateWithOAuth2() {
-      let { identification, password } = this.getProperties('identification', 'password');
-      this.get('session').authenticate('authenticator:oauth2', identification, password)
-        .then(() => {
-          this.get('rememberMe') && this.set('session.store.cookieExpirationTime', 60 * 60 * 24 * 14);
-        })
-        .catch((reason) => {
-          this.set('errorMessage', reason.error);
-        });
+    async authenticateWithOAuth2() {
+      try {
+        let { identification, password } = this;
+        await this.get('session').authenticate('authenticator:oauth2', identification, password);
+
+        if (this.rememberMe) {
+          this.get('session').set('store.cookieExpirationTime', 60 * 60 * 24 * 14);
+        }
+      } catch (response) {
+        let responseBody = await response.clone().json();
+        this.errorMessage = responseBody.error;
```

Comment:
this looks quite different than the previous implementation 🤔

Reply:
Yes, this is actually broken on `master` currently…

Reply:
then shouldn't we fix this in a dedicated PR?

I find this PR somewhat hard to review because it's changing so many things at the same time

Reply:
Yes, makes sense: #2138